wallet/src/de/schildbach/wallet/WalletApplication.java

```diff
@@ -132,6 +132,11 @@ public void onCreate() {
                 //noinspection ResultOfMethodCallIgnored
                 walletFile.delete();
                 config.clear(false);
```

Comment:
There is an issue with this `config.clear(false)` being here:

If a user does wipe the wallet, they will go to the onboarding process and create or recover a wallet.  When the app is restarted, this section of code will delete the wallet file.  The wallet file was recreated during the onboarding process, but ensureWipe is still true.  So here, the wallet file is deleted and ensureWipe = false.  

later in the app loading process, the wallet file will not be found and then the backup file is used to restore the wallet and this requires a blockchain reset.  
```
2019-12-03 13:48:28.485 14725-14725/hashengineering.dash.wallet.beta E/WalletApplication: [main] problem loading wallet
    java.io.FileNotFoundException: /data/user/0/hashengineering.dash.wallet.beta/files/wallet-protobuf (No such file or directory)
```

`config.clear(false);` should be placed at the end of the onboarding process, to clear this state.

 After the wipe() method is called, the wallet file is deleted.  Then the app is restarted.  When onCreate() is executed, the walletFile does not exist, so the `config.clear(false);` is not called to clear the state of ensureWipe.

Perhaps this block of code doesn't need to be here at all since the delete file is in wipe() and config.clear() can go to the end of onboarding.  The blockchain file is deleted in wipe().

Reply:
Hmm... interesting observation. The idea was that the app is being immediately restarted after wipe (by calling `ProcessPhoenix.triggerRebirth(context);`) and mentioned code from the `WalletApplication.onCreate` method, that ensures removing wallet file, is called before the user is able to create/restore the wallet using onboarding flow.
I will double-check this.

Reply:
I understand your reasoning, but the **wipe()** method has already deleted the wallet file.  I like the idea of keeping the delete wallet function in the **wipe()** method and not here.  It is a bit scary to see a **walletFile.delete()** near the code that is executed each time the app is opened up.

Reply:
Ok, I will re-validate this code.
It was added since in the past I had the issues with deleting the wallet file before restarting the wallet. The `walletFile.delete()` from `wipe()` returned `false` and file wasn't deleted. I thought it was caused by the use of this file in `BlockchainServiceImpl` but this set of instructions:
```
resetBlockchain();
wallet.shutdownAutosaveAndWait();
stopBlockchainService();
```
should avoid such a failure.
Anyways, I think the `wipe()` function became so susceptible that it should be developed and tested within a dedicated task...

Reply:
Ok, we will do a separate PR on wipe().